processing/src/main/java/io/druid/query/topn/BaseTopNAlgorithm.java

```diff
@@ -196,10 +199,6 @@ public BaseArrayProvider(
       ignoreAfterThreshold = false;
       ignoreFirstN = 0;
       keepOnlyN = dimSelector.getValueCardinality();
-
-      if (keepOnlyN < 0) {
-        throw new UnsupportedOperationException("Cannot operate on a dimension with no dictionary");
-      }
```

Comment:
Restored this after splitting the base topN algorithm to have separate "with cardinality" and "unknown cardinality" paths